lang/java/compiler/src/main/java/org/apache/avro/compiler/specific/SpecificCompiler.java

```diff
@@ -1172,6 +1167,40 @@ private static String generateMethodName(Schema schema, Field field, String pref
     return methodBuilder.toString();
   }
 
+  /**
+   * CamelCase the incoming string, using underscores as the word hints
+   * @param s string to camel-case. Must contain only [a-zA-Z0-9_]
+   * @param upper should the first letter be capitalized
+   */
+  static String camelize(String s, boolean upper) {
+    if (StringUtils.contains(s, '_')) {
+      // use underscores as hints where the capitals go
+      StringBuilder builder = new StringBuilder();
+      for (String part : StringUtils.split(s, '_')) {
+        builder.append(StringUtils.capitalize(StringUtils.lowerCase(part)));
```

Comment:
'StringUtils.lowerCase ()' will have better performance if placed outside the loop.